doc/rst/developer/bestPractices/Deprecation.rst

```diff
@@ -1,9 +1,46 @@
 Chapel's Deprecation Policy
 ===========================
 
-Chapel's general policy for deprecation is to:
+When removing a feature in Chapel, the feature is to be
+deprecated and supported for at least 1 release cycle.
 
-- Provide a compiler warning (or error) when a deprecated feature is used
-- Continue to document and support the deprecated feature for 1 release cycle
-  - The documentation should contain a pointer to the preferred feature
-  - This holds true for modules as well
+Major feature removals can be supported for more than 1 release cycle.
+
+Non-public interface features can be removed or modified without deprecation.
+
+This is the deprecation process:
+
+1. Provide a compiler warning (or error, if necessary) when a deprecated feature is used
```

Comment:
Maybe "Provide a compiler, runtime, or module warning (or error, if necessary) to be printed when a deprecated feature is used."  See for example 8a3dec34c4, which added such a deprecation warning to runtime/src/config.c.

Reply:
Good point. I'm going to generalize my example to just the text rather than attempt to demonstrate the matrix of cases across {warning, error} + {modules, compiler, runtime} + {runtime, compile-time}.